lms/djangoapps/courseware/tests/test_access.py

```diff
@@ -111,20 +146,27 @@ def test__has_access_string(self):
 
         self.assertRaises(ValueError, access._has_access_string, user, 'not_staff', 'global')
 
-    def test__has_access_error_desc(self):
+    @ddt.data(
+        ('load', False, True, True),
+        ('staff', False, True, True),
+        ('instructor', False, False, True)
+    )
+    @ddt.unpack
+    def test__has_access_error_desc(self, action, expected_student, expected_staff, expected_instructor):
         descriptor = Mock()
 
-        self.assertFalse(access._has_access_error_desc(self.student, 'load', descriptor, self.course.course_key))
-        self.assertTrue(access._has_access_error_desc(self.course_staff, 'load', descriptor, self.course.course_key))
-        self.assertTrue(access._has_access_error_desc(self.course_instructor, 'load', descriptor, self.course.course_key))
-
-        self.assertFalse(access._has_access_error_desc(self.student, 'staff', descriptor, self.course.course_key))
-        self.assertTrue(access._has_access_error_desc(self.course_staff, 'staff', descriptor, self.course.course_key))
-        self.assertTrue(access._has_access_error_desc(self.course_instructor, 'staff', descriptor, self.course.course_key))
-
-        self.assertFalse(access._has_access_error_desc(self.student, 'instructor', descriptor, self.course.course_key))
-        self.assertFalse(access._has_access_error_desc(self.course_staff, 'instructor', descriptor, self.course.course_key))
-        self.assertTrue(access._has_access_error_desc(self.course_instructor, 'instructor', descriptor, self.course.course_key))
+        self.assertEquals(
+            bool(access._has_access_error_desc(self.student, action, descriptor, self.course.course_key)),
+            expected_student
+        )
+        self.assertEquals(
+            bool(access._has_access_error_desc(self.course_staff, action, descriptor, self.course.course_key)),
+            expected_staff
+        )
+        self.assertEquals(
+            bool(access._has_access_error_desc(self.course_instructor, action, descriptor, self.course.course_key)),
+            expected_instructor
+        )
```

Comment:
Awesome refactoring with `ddt`.  

Consider consolidating even further by eliminating duplicate code.  For example:

```
   for (user, expected_response) in (
        (self.student, expected_student),
        (self.course_staff, expected_staff),
        (self.course_instructor, expected_instructor)
   ):
        self.assertEquals(
            bool(access._has_access_error_desc(user, action, descriptor, self.course.course_key)),
            expected_response
        )
```